source/common/router/router.cc

```diff
@@ -1045,10 +1045,21 @@ void Filter::onStreamMaxDurationReached(UpstreamRequest& upstream_request) {
 
   callbacks_->streamInfo().setResponseFlag(
       StreamInfo::ResponseFlag::UpstreamMaxStreamDurationReached);
+  const auto& stream_info = callbacks_->streamInfo();
   // sendLocalReply may instead reset the stream if downstream_response_started_ is true.
   callbacks_->sendLocalReply(
-      Http::Code::RequestTimeout, "upstream max stream duration reached", modify_headers_,
-      absl::nullopt, StreamInfo::ResponseCodeDetails::get().UpstreamMaxStreamDurationReached);
+      stream_info.downstreamTiming().has_value() &&
```

Comment:
nit: can we break this boolean out of the sendLocalReply call for readability?

Reply:
done